tasks/managed/close-advisory-issues/close-advisory-issues.yaml

```diff
@@ -0,0 +1,100 @@
+      type: string
+  workspaces:
+    - name: data
+      description: The workspace where the snapshot spec json file resides
+  steps:
+    - name: close-issues
+      image: quay.io/konflux-ci/release-service-utils:0f82be4be43294b6a96846d87ef7f7c0b9e34267
+      env:
+        - name: ACCESS_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: todo
+              key: TODO
+      script: |
+        #!/usr/bin/env bash
+        # We do not set -x here because it would leak the ACCESS_TOKEN
+
+        ISSUE_TRACKERS='{
+            "Jira": {
+                "api": "rest/api/2/issue",
+                "servers": [
+                    "issues.redhat.com",
+                    "jira.atlassian.com"
+                ]
+            },
+            "bugzilla": {
+                "api": "rest/bug",
+                "servers": [
+                    "bugzilla.redhat.com"
+                ]
+            }
+        }'
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No data JSON was provided."
+            exit 1
+        fi
+
+        RC=0
+
+        NUM_ISSUES=$(jq -cr '.releaseNotes.issues.fixed | length' "${DATA_FILE}")
+        for ((i = 0; i < NUM_ISSUES; i++)); do
+            issue=$(jq -c --argjson i "$i" '.releaseNotes.issues.fixed[$i]' "${DATA_FILE}")
+            server=$(jq -r '.source' <<< "$issue")
+            if [ "$server" != "issues.redhat.com" ] ; then
+                echo "This task currently only supports closing issues on issues.redhat.com"
+                echo "Skipping issue $issue as it is on $server"
+                continue
+            fi
+            echo "Closing issue $issue on server $server"
+
+            CURL_ARGS=(
+              -H "Authorization: Bearer $ACCESS_TOKEN"
+              --retry 3
+              --fail
+            )
+
+            API=$(jq -r '.[] | select(.servers[] | contains("'"$server"'")) | .api' <<< "$ISSUE_TRACKERS")
+            API_URL="https://$(jq -r '.source' <<< "$issue")/${API}/$(jq -r '.id' <<< "$issue")/transitions"
+            # Get the Closed transition id. This varies per Jira project
+            if ! CLOSED_ID="$(curl "${CURL_ARGS[@]}" "${API_URL}" \
+              | jq -r '.transitions[] | select(.name=="Closed") | .id')"; then
+                echo "Error: failed to fetch the closed state id for issue $issue."
+                RC=1
+                continue
+            fi
+
+            # Close the issue
+            CLOSE_COMMENT="Fixed in Konflux Advisory $(params.advisoryUrl)"
+            if ! curl "${CURL_ARGS[@]}" -XPOST --data \
+              '{"transition":{"id":"'"$CLOSED_ID"'"},"update":{"comment":[{"add":{"body":"'"$CLOSE_COMMENT"'"}}]}}' \
+              -H "Content-Type: application/json" "${API_URL}"; then
+                echo "Error: failed to close issue $issue."
```

Comment:
What happens if the jira is already closed? This should just pass I think. For idempotency. Does it?

Reply:
It adds this to the activity for example
```
John Bieren made changes - 2025/02/07 12:26 PM
Status	Original: Closed [ 6 ]	New: Closed [ 6 ]
```
So, it doesn't actually do anything (which is good) but it does update the "updated" timestamp on the jira. The comment would also be reposted. 
I can curl the issue and do `jq '.fields.status.name'` and if = Closed, `continue`. What do you think? It adds another curl call but sounds like it can be worth it for idempotent's sake

Reply:
I think the current version is acceptable, but it would be nice to add the check and skip it.

Reply:
will add it

Reply:
added